### ISSUE TITLE
Question: Any possible benefit from switching to Cython?

### ISSUE THREAD

**Issue by stefanos82** [mesonbuild/meson#5987]
@jpakkane After watching your [CPPCon presentation](https://www.youtube.com/watch?v=34KzT2yvQuM), I wanted to ask you.

Many users believe Meson should be rewritten in static typed programming language, such as C, C++, or Rust; by the way, I have found these opinions online via searching.

Since Cython can be used to either emit a C extension for Python **OR** to embed Python in an executable, do you think Meson could benefit out of all this?

Thank you very much.

P.S.: The following example below is a demonstration about how Cython could benefit anyone who needs performance improvements that match those of C language.

Those who don't know how to do such thing with Cython, let me demonstrate a silly example:

Call this `helloworld.pyx` (the comment plays role for Cython in the following excerpt):
```py
#cython: language_level=3

print("Hello World")
```
Execute the following command: `cython helloworld.pyx --embed`

This will generate a `.c` file with the name of our `.pyx` file, i.e.: `helloworld.c`

Then compile the code as follows: `gcc -O2 helloworld.c -o helloworld $(python-config --libs --cflags --ldflags)`

Here I assumed that the default compiler is GCC; feel free to choose whatever you use or feel comfortable with its usage.

In other words, anywhere Python is found and is accompanied by a C compiler, you can produce an executable and maintain all the Python Standard Library dependency you are worrying about.

Lastly, strip your executable file from unnecessary data with `strip -s helloworld`.

Congrats, you have yourself compiled an executable that runs Python code embedded in it.

**Comment by eli-schwartz** [comment]
I'm unaware of people complaining that meson is too slow. Every complaint I've heard about the meson implementation language was about the use of python incurring a dependency which may be hard to fulfill, especially in bootstrapping scenarios.

Cython would not be any help at all, since it would still require python!

**Comment by stefanos82** [comment]
@eli-schwartz yeah, for some reason I forgot how clear they were (in @jpakkane's presentation) about converting the project to either C or Perl.

It's a shame because it's an incredible project.

**Comment by jpakkane** [comment]
The main reasons for slowdown seem to be running compiler checks and looking up dependencies with external binaries, especially on Windows. Cython does not really help with that.

**Comment by stefanos82** [comment]
Very nice, it's good to know.

Closing this ticket.
